moq-arch.md

```diff
@@ -2,87 +2,99 @@
+such as detecting available bandwidth, fragmentation and 
+reassembly, resiliency, congestion control and prioritization of
 data delivery based on data lifetime and importance of data. It is
 designed to be NAT and firewall traversal friendly and can be fronted
-with load balancers. Objects are named such that it is unique for the
-relay/delivery network and scoped to an application. Subscriptions can
-include a form of wildcarding to the named object.
+with load balancers. 
+
+The Relays are arranged in a logical tree (as shown below) where, 
+for a given application, there is an origin Relay at root of the tree 
+that controls the namespace. Publish messages are sent towards the 
+root of the tree and down the path of any subscribers to that 
+named object. QuicR is designed to make it easy to implement relays 
+so that fail over could happen between relays with minimal impact to 
+the clients and relays can redirect a client to a different relay.
+
+!--
+~~~ascii-art 
+              ┌────────────┐
+              │            │
+              │            ▼
+              │       ┌────────┐
+              │   ▬ ▬▶│Relay-0 │ ◀▬▬ ▬▬ ▬▮
+          pub │  ▮    │ Orgin  ├┐        ▮
+              │  ▮     ────────┘│        ▮
+              │  ▮ sub          │        ▮ sub
+              │  ▮          pub │        ▮
+              │  ▮              │        ▮
+         ┌───────▮┐ ◀▬▮         │  ┌─────▮──┐
+     ┌──▶│ Relay-1│   ▮         └─▶│ Relay-2│◀▮▮
+     │   └─────┬──┘   ▮             ▲───┬───┘  ▮
+ pub │         │      ▮ sub     sub ▮   │      ▮ sub
+     │      pub│      ▮             ▮   │      ▮
+    ┌┴─────┐   │ ┌────▮─┐     ┌─────▮┐  │  ┌───▮──┐
+    │Alice │   └▶│ Bob  │     │ Carl │◀─┴─▶│Derek │
+    └──────┘     └──────┘     └──────┘ pub └──────┘
+~~~
+Figure: QuicR Delivery Tree
+!--
 
 The design supports sending media and other named objects between a set
 of participants in a game or video call with under a hundred
 milliseconds of latency and meets the needs of web conferencing
 systems. The design can also be used for large scale streaming to
 millions of participants with latency ranging from a few seconds to
-under a  hundred milliseconds based on applications needs. It can
+under a hundred milliseconds based on applications needs. It can
 also be used as low latency publish/subscribe system for real time
 systems such as messaging, gaming, and IoT.
 
-In the simplest case, a web conferencing application could use a single
-relay to forward packets between users in a video conference. However a
-more typical scenario would have a delivery network made of multiple
-relays spread across several points of presence. QuicR is designed to
-make it easy to implement relays so that fail over could happen between
-relays with minimal impact to the clients and relays can redirect a
-client to a different relay.
+QuicR is pronounced something close to
+“quicker” but with more of a pirate "arrrr" at the end.
```

Comment:
There seems to be something wrong on the smart quoates on the arrr (like using unicode) - I thought I had fixed this in some other pull and now I am worried I did not merge that or something. Not a problem with this PR but I need to figure out where that went

Reply:
I fixed this